provider/image.go

```diff
@@ -457,29 +464,43 @@ func getRegistryAddr(serverName, imgName string) (string, error) {
 
 	} else {
 		// check if the provider registry server starts with https://
-		if strings.Contains(serverName, "https://") {
+		if strings.HasPrefix(serverName, "https://") {
 			serverAddr = serverName
 		} else {
-			// TODO: this is where we would default to Docker if we wanted to do so.
-			return "", errors.Errorf("invalid registry server name: %s", serverName)
+			// courtesy add the prefix so user does not have to explicitly do so
+			serverAddr = "https://" + serverName
 		}
 	}
 	return serverAddr, nil
 }
 
+func getRegistryAddrFromImage(imgName string) (string, error) {
+	named, err := reference.ParseNamed(imgName)
+	if err != nil {
+		msg := fmt.Errorf("error: %s. This provider requires all image names to be fully qualified.\n"+
+			"For example, if you are attempting to push to Dockerhub, prefix your image name with `docker.io`:\n\n"+
+			"`docker.io/repository/image:tag`", err)
+		fmt.Println(msg)
+		return "", err
+	}
+	addr := reference.Domain(named)
+	return addr, nil
+
+}
```

Comment:
Can you add a unit test for the expected cases here?

Reply:
Done!